leaderelection/src/test/java/io/scalecube/leaderelection/LeaderElectionIT.java

```diff
@@ -0,0 +1,118 @@
+package io.scalecube.leaderelection;
+
+import static org.junit.Assert.assertTrue;
+
+import io.scalecube.cluster.Cluster;
+import io.scalecube.cluster.ICluster;
+import io.scalecube.transport.Address;
+
+import com.google.common.base.Throwables;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class LeaderElectionIT {
+
+  Executor exec = Executors.newCachedThreadPool();
+  @Test
+  public void testAllNodesFollowOneLeader() {
+    ICluster seed = Cluster.joinAwait();
+    LeaderElection seedLeader = createLeaderElection(seed);
+    List<LeaderElection> leaders = createCluster(seed);
+
+    // waiting for a leader to be selected
+    awaitSeconds(30);
+
+    Address seedLeaderAddress = seedLeader.leader();
+    
+    for (LeaderElection m : leaders) {
+      Address memberLeaderAddress = m.leader();
+      assertTrue("Expected leader address " + seedLeaderAddress + ", but actual: " + memberLeaderAddress,
+          memberLeaderAddress.equals(seedLeaderAddress));
+    }
+    
+    // shutdown everything
+    for (LeaderElection m : leaders) {
+      m.cluster().shutdown();
+    }
+    seed.shutdown();
+  }
+
+  
+  @Test
+  public void testLeaderDiesAndNewLeaderIsElected() {
+    ICluster seed = Cluster.joinAwait();
+    LeaderElection seedLeader = createLeaderElection(seed);
+    List<LeaderElection> leaders = createCluster(seed);
+
+    // waiting for a leader to be selected
+    awaitSeconds(30);
+
+    killTheLeader(leaders,seedLeader.leader());
+    
+    // waiting for a leader to be selected
+    awaitSeconds(30);
+
+    Address newLeader = seedLeader.leader();
+    
+    for (LeaderElection m : leaders) {
+       Address memberLeaderAddress = m.leader();
+       assertTrue("Expected leader address " + newLeader + ", but actual: " + memberLeaderAddress,
+       memberLeaderAddress.equals(newLeader));
+    }
+    
+    // shutdown everything
+    for (LeaderElection m : leaders) {
+      m.cluster().shutdown();
+    }
+    seed.shutdown();
+  }
+
+  
+  private void killTheLeader(List<LeaderElection> leaders, Address leader) {
+    LeaderElection rm = null;
+    for (LeaderElection m : leaders) {
+      if(leader.equals(m.cluster().address())) {
+        m.cluster().shutdown();
+        rm = m;
+        break;
+      }
+    }
+    leaders.remove(rm);
+  }
+
+
+  private List<LeaderElection> createCluster(ICluster seed) {
+    List<LeaderElection> leaders = new ArrayList<>();
+    for (int i = 0 ; i < 50; i++) {
+      exec.execute(new Runnable() {
+        @Override
+        public void run() {
+          ICluster member = Cluster.joinAwait(seed.address());
+          leaders.add(createLeaderElection(member));  
+        }
+      });
+    }
+    return leaders;
+  }
+
+    
```

Comment:
Here required more tests for different adverse conditions. For example:

* Test case when you select leader then disable network from/to this leader, check that in the rest of the cluster new leader elected, also what will happen with this current leader? then enable network againfor this leader and check that this previous leader will agree on new one.

* Split cluster on two partitions one is quorum and another isn't. Check that quorum part will select leader, what will be with non quorum? then join cluster again and check that leader is stable.

* Try to fail some mebers in the process of leader election and check that leader is still elected correctly

* Come up with some more creative test cases